demo/pubsub_forwarder/pubsub_forwarder.c

```diff
@@ -0,0 +1,86 @@
+//
+// This software is supplied under the terms of the MIT License, a
+// copy of which should be located in the distribution where this
+// file was obtained (LICENSE.txt).  A copy of the license may also be
+// found online at https://opensource.org/licenses/MIT.
+//
+
+//
+// Forwarder example based on https://github.com/C-o-r-E/nng_pubsub_proxy
+//
+// This example shows how to use raw sockets to set up a forwarder or proxy for pub/sub.
+// 
+// An example setup for running this example would involve the following:
+// 
+//  - Run this example binary (in the background or a terminal, etc)
+//  - In a new terminal, run `nngcat --sub --dial "tcp://localhost:3328" --quoted`
+//  - In a second terminal, run `nngcat --sub --dial "tcp://localhost:3328" --quoted`
+//  - In a third terminal, run `for n in $(seq 0 99); do nngcat --pub --dial "tcp://localhost:3327" --data "$n"; done`
+//
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <nng/nng.h>
+#include <nng/protocol/pubsub0/sub.h>
+#include <nng/protocol/pubsub0/pub.h>
+
+#define PROXY_FRONT_URL "tcp://localhost:3327"
+#define PROXY_BACK_URL "tcp://localhost:3328"
+
+void panic_on_error(int should_panic, const char* format, ...)
+{
+    if (should_panic)
+    {
+        va_list args;
+        va_start(args, format);
+        vprintf(format, args);
+        va_end(args);
+        exit(EXIT_FAILURE);
+    }
+}
+
+int main()
+{
+    nng_socket sock_front_end =   NNG_SOCKET_INITIALIZER;
+    nng_socket sock_back_end =    NNG_SOCKET_INITIALIZER;
+    int ret = 0;
+
+    /*
```

Comment:
This is a super silly nit, but I don't use these style of block comments anywhere else in this code base.

If you're willing, I'd appreciate if you could convert them to C++ style just for consistency.

I won't hold the PR up for this, because at least on the face of it, it is very nice and I'd like to see it land before the next release, but I will wait to give you a chance to respond (and possibly update) before I merge.

Reply:
No problem. Will change them to match